generic-examples/kafka/README.md

```diff
@@ -1,30 +1,52 @@
+cluster before starting the example.
+
+## Additional Requirements for running the examples
+**A Kafka broker**: handles the storage and passing of messages.
 
 ## Authentication to Kafka
 
-We have a dedicated section to show [how to authenticate Camel K to Kafka](./sasl_ssl/).
+To use Kafka with authentication, we have a dedicated section to show [how to authenticate Camel K to Kafka](./sasl_ssl/). \
+For a simple use case without client authentication, continue with this guide.
+
+## Understanding the example
+- [SampleKafkaConsumer.java](./SampleKafkaConsumer.java): contains a route that reads message from a kafka topic and logs the message
+- [application.properties](./application.properties): holds properties required to connect to kafka broker and read from topic.
 
 ## Simple usage
 
-To run this example first set-up Kafka on your k8s cluster.
-A convenient way to do so is by using the Strimzi project, if you are using minikube follow these instructions at https://strimzi.io/quickstarts/minikube/
+To run this example, first set-up Kafka on your k8s cluster.
+A convenient way to do so is by using the Strimzi project. Visit https://strimzi.io/quickstarts/ for set-up instructions. For the instructions on the linked site, it will suffice to only apply the Strimzi installation file and provision the kafka cluster.
 
-For convenience create a configmap to contain the properties:
+IMPORTANT: The `kafka.host` value in `application.properties` needs to be set to the CLUSTER-IP address of the my-cluster-kafka-bootstrap service in the kafka namespace. To do this run:
 ```
-kubectl create configmap kafka.props  --from-file=examples/kafka/application.properties
+kafkaip=`kubectl get svc/my-cluster-kafka-bootstrap -n kafka -ojsonpath="{.spec.clusterIP}"`; sed -i "/kafka\.host/s/<.*>/$kafkaip/g" application.properties
+```
+
+Create a configmap to contain the properties:
+```
+kubectl create configmap kafka.props  --from-file=application.properties
 ```
 
-IMPORTANT: The kafka.host value in application.properties needs to be set to the CLUSTER-IP address of the my-cluster-kafka-bootstrap service in the kafka namespace:
- `kubectl get services -n kafka | grep my-cluster-kafka-bootstrap | awk '/[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/ {print $3}'`
+
 
 Finally run this sample using the command:
 ```
-./kamel run examples/kafka/SampleKafkaConsumer.java --config=configmap:kafka.props
+kamel run SampleKafkaConsumer.java --config=configmap:kafka.props --dev
 ```
 
-To create messages to be read use the producer command from the Strimzi page:
+To create messages to be read, use the producer command from the Strimzi page. Run in another terminal:
+```
+kubectl -n kafka run kafka-producer -ti --image=quay.io/strimzi/kafka:0.30.0-kafka-3.2.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic my-topic
```

Comment:
We rarely use `--rm=true`. It can be simpler: `--rm`
```
kubectl -n kafka run kafka-producer -it --rm --image=quay.io/strimzi/kafka:0.30.0-kafka-3.2.0 --restart=Never -- bin/kafka-console-producer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic my-topic
```